docs/index.rst

```diff
@@ -111,54 +113,75 @@ function.
 
 .. code-block:: python
 
+    import flask
+    from flask_login import login_user
+
     @app.route('/login', methods=['GET', 'POST'])
     def login():
         # Here we use a class of some kind to represent and validate our
         # client-side form data. For example, WTForms is a library that will
         # handle this for us, and we use a custom LoginForm to validate.
         form = LoginForm()
+
         if form.validate_on_submit():
             # Login and validate the user.
             # user should be an instance of your `User` class
             login_user(user)
 
             flask.flash('Logged in successfully.')
 
-            next = flask.request.args.get('next')
+            next_url = form.next_url.data
             # is_safe_url should check if the url is safe for redirects.
             # See http://flask.pocoo.org/snippets/62/ for an example.
-            if not is_safe_url(next):
+            if not is_safe_url(next_url):
                 return flask.abort(400)
 
-            return flask.redirect(next or flask.url_for('index'))
+            return flask.redirect(next_url or flask.url_for('index'))
+
+        # Pass the next URL along through the Login form.
+        form.next_url.data = flask.request.args.get('next')
         return flask.render_template('login.html', form=form)
 
-*Warning:* You MUST validate the value of the `next` parameter. If you do not,
+*Warning:* You MUST validate the value of the `next_url` parameter. If you do not,
 your application will be vulnerable to open redirects. For an example
 implementation of `is_safe_url` see `this Flask Snippet`_.
 
+See the `Flask-WTF documentation on creating forms`_ for examples on how to create
+your `LoginForm` class and corresponding template. Make sure your template
+includes the `{{ form.csrf_token }}` field, otherwise it will fail to validate.
+Also make sure to include a hidden `next_url` field so that it can be used to pass
```

Comment:
Leave this is `next` not `next_url`. Only rename the variable used in example user-land view to `next_url` but keep the url arg named `next`.